llpc/tool/amdllpc.cpp

```diff
@@ -577,6 +581,53 @@ static Result initCompileInfo(CompileInfo *compileInfo) {
   return Result::Success;
 }
 
+static Error fixupRtState(RtState &rtState, std::vector<char> &shaderLibraryStorage) {
+  if (rtState.gpurtOverride || rtState.rtIpOverride) {
+    // Trying to compile a .pipe file that was dumped from a driver with an explicit shader library override setting
+    // may have unexpected results. Complain loudly to make sure the user knows what they're doing.
+    if (!GpuRtUseDumped.getNumOccurrences() && GpuRtLibrary.empty()) {
+      return createResultError(Result::ErrorInvalidValue,
+                               "The .pipe file explicitly sets gpurtOverride and/or rtIpOverride to true. You must "
+                               "explicitly choose -gpurt-use-dumped=true/false.");
+    }
+
+    if (!GpuRtUseDumped) {
+      rtState.gpurtOverride = false;
+      rtState.rtIpOverride = false;
+    }
+  }
+
+  if (GpuRtUseDumped)
+    rtState.gpurtOverride = true;
+  else
+    rtState.gpurtShaderLibrary = {};
+
+  if (!GpuRtLibrary.empty()) {
+    if (GpuRtUseDumped)
+      return createResultError(Result::ErrorInvalidValue, "Cannot use both -gpurt-use-dumped and -gpurt-library");
+
+    FILE *gpurtFile = fopen(GpuRtLibrary.c_str(), "rb");
+    if (!gpurtFile)
+      return createResultError(Result::ErrorUnavailable, Twine("Failed to open input file: ") + GpuRtLibrary);
+    auto closeInput = make_scope_exit([gpurtFile] { fclose(gpurtFile); });
+
+    fseek(gpurtFile, 0, SEEK_END);
+    shaderLibraryStorage.resize(ftell(gpurtFile));
+    fseek(gpurtFile, 0, SEEK_SET);
+    size_t nread = fread(shaderLibraryStorage.data(), 1, shaderLibraryStorage.size(), gpurtFile);
+    fclose(gpurtFile);
```

Comment:
Double `fclose`? You are already closing with the RAII `closeInput` helper.

Reply:
You're right. I'll fix it, but there was a (apparently false positive) cppcheck report that is likely to reappear :(